IJReachability/IJReachability/IJReachability.swift

```diff
@@ -22,55 +34,113 @@ public class IJReachability {
     */
     public class func isConnectedToNetwork() -> Bool {
         
-        var zeroAddress = sockaddr_in(sin_len: 0, sin_family: 0, sin_port: 0, sin_addr: in_addr(s_addr: 0), sin_zero: (0, 0, 0, 0, 0, 0, 0, 0))
+        var zeroAddress = sockaddr_in()
         zeroAddress.sin_len = UInt8(sizeofValue(zeroAddress))
         zeroAddress.sin_family = sa_family_t(AF_INET)
-        
-        let defaultRouteReachability = withUnsafePointer(&zeroAddress) {
-            SCNetworkReachabilityCreateWithAddress(nil, UnsafePointer($0)).takeRetainedValue()
+
+        guard let defaultRouteReachability = withUnsafePointer(&zeroAddress, {
+            SCNetworkReachabilityCreateWithAddress(nil, UnsafePointer($0))
+        }) else {
+            return false
         }
         
-        var flags: SCNetworkReachabilityFlags = 0
+        var flags : SCNetworkReachabilityFlags = []
+        
         if SCNetworkReachabilityGetFlags(defaultRouteReachability, &flags) == 0 {
```

Comment:
I get the error: Binary operator '==' cannot be applied to operands of type 'Bool' and 'Int' on beta 5. Error went away by replacing 0 by 'false', or when replacing the code to:  

if !SCNetworkReachabilityGetFlags(defaultRouteReachability, &flags) { return false }